### PR TITLE
Refactor(plugins): Move internal cv_client code to PyAVD

### PR DESCRIPTION
**Pull request by ClausHolbechArista**
## Change Summary

Move internal cv_client code to PyAVD

### User Checklist

- N/A

### Repository Checklist

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contribution/overview.html) document.
- [x] My change requires a change to the documentation and documentation have been updated accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly. (check the box if not applicable)

### CODE REVIEW

File: python-avd/pyproject.toml [hunk 1]
```diff
@@ -75,7 +75,7 @@ branch = true
 
 [tool.coverage.report]
 show_missing = true
-fail_under = 89
+fail_under = 88
```
Comment (gmuloc) [inline python-avd/pyproject.toml]: can we not just ignore the cv api files?

Comment (ClausHolbechArista) [reply python-avd/pyproject.toml]: we could. But it _is_ untested code.

File: ansible_collections/arista/avd/plugins/action/cv_workflow.py [hunk 2]
```diff
@@ -84,6 +94,9 @@ def run(self, tmp=None, task_vars=None):
         result = super().run(tmp, task_vars)
         del tmp  # tmp no longer has any effect
 
+        if not HAS_PYAVD:
+            raise AnsibleActionFail("The Python library 'pyavd' was not found. Install using 'pip3 install'.")
```
Comment (gmuloc) [inline ansible_collections/arista/avd/plugins/action/cv_workflow.py:98]: pip3?

**Review by gmuloc — APPROVED** [review]
LGTM once error message is changed to remove mention to `pip` as we discuessed and CI passes.

**Pull request merged**